### ISSUE TITLE
Redundant local name for modified imports

### ISSUE THREAD

**Issue by scop** [timothycrosley/isort#1281]
isort 5.0.4 changes this
```python
import urllib.parse as parse
```
...to this:
```python
from urllib import parse as parse
```

I think `as parse` is redundant and should not be emitted, but just
```python
from urllib import parse
```
FWIW pylint will issue `useless-import-alias` for the duplication.

**Comment by jdufresne** [comment]
More generally, perhaps isort should remove useless aliases whether isort generated them or not:

**input**:

```py
from datetime import datetime as datetime

print(datetime.now())
```

**current output**: unchanged

**expected output**:

```py
from datetime import datetime

print(datetime.now())
```

**Comment by scop** [comment]
I agree. But for completeness, I've seen documented conventions where those "useless" aliases are actually used to mark an import being part of an API. So e.g. for a module that doesn't use `foo` itself at all, `from bar import foo as foo` with the alias serves to note that the import isn't accidental. I can't locate a reference to such a convention at the moment, nor do I actually personally find it a good one, but thought I'd mention it, FWIW.

**Comment by timothycrosley** [comment]
This is fixed in develop and will be released in the 5.1.0 minor release slated for release by end of month at the latest. As part of this a new `remove_redundent_aliases` [config option](https://timothycrosley.github.io/isort/docs/configuration/options/#remove-redundant-aliases) has been added - though no alias will now show up by default either way if there was not one present originally.

Thanks!

~Timothy

**Comment by timothycrosley** [comment]
This has now been deployed to PyPI in the [5.1.0 release](https://timothycrosley.github.io/isort/CHANGELOG/#510-july-14-2020), thanks again!

~Timothy
